pkg/commands/git_commands/gh.go

```diff
@@ -0,0 +1,128 @@
+package git_commands
+
+import (
+	"encoding/json"
+	"fmt"
+	"strings"
+
+	"github.com/jesseduffield/lazygit/pkg/commands/hosting_service"
+	"github.com/jesseduffield/lazygit/pkg/commands/models"
+)
+
+type GhCommands struct {
+	*GitCommon
+}
+
+func NewGhCommand(gitCommon *GitCommon) *GhCommands {
+	return &GhCommands{
+		GitCommon: gitCommon,
+	}
+}
+
+// https://github.com/cli/cli/issues/2300
+func (self *GhCommands) BaseRepo() error {
+	return self.cmd.New("git config --local --get-regexp .gh-resolved").StreamOutput().Run()
+
+}
+
+// Ex: git config --local --add "remote.origin.gh-resolved" "jesseduffield/lazygit"
+func (self *GhCommands) SetBaseRepo(repository string) (string, error) {
+	return self.cmd.NewShell(fmt.Sprintf("git config --local --add \"remote.origin.gh-resolved\" \"%s\"", repository)).RunWithOutput()
+}
+
+func (self *GhCommands) prList() (string, error) {
+	return self.cmd.NewShell("gh pr list --limit 100 --state all --json state,url,number,headRefName,headRepositoryOwner").RunWithOutput()
```

Comment:
The list is 100, so if you have an old PR, you may not be able to find it.
It may be better to find the corresponding PR by local branch name.
but for now I think this is ok